packages/backend/src/open_payments/payment/incoming_remote/service.ts

```diff
@@ -120,10 +113,23 @@ async function getGrant(
 
   if (existingGrant) {
     if (existingGrant.expired) {
-      // TODO https://github.com/interledger/rafiki/issues/795
-      const errorMessage = 'Grant access token expired'
-      deps.logger.error({ grantOptions }, errorMessage)
-      return RemoteIncomingPaymentError.ExpiredGrant
+      if (!existingGrant.authServer) {
+        return RemoteIncomingPaymentError.UnknownAuthServer
```

Comment:
This _shouldn't_ happen
https://github.com/interledger/rafiki/blob/bbebbdf987b00b6256e239ef30497a21dca3ceb2/packages/backend/src/open_payments/grant/service.ts#L65-L73
Should we instead throw? Or have `grantService.get` return a type where `authServer` is required?

Reply:
> Or have `grantService.get` return a type where authServer is required?

I like this as a general pattern

Reply:
_maybe_ we could just have a separate method even?

Reply:
Returning a different type where `authServer` is required is not that straight forwards due to the `PaginationQueryBuilder` on the `BaseModel`. I think we should just throw.